Added fix for Issue 59: Instructions to search for items

mehulsr opened the pull request:
Fixes #59 
Trying to figure out how to add link to README.md

TimMoore commented:
@mehulsr here is a pointer for linking to the README: https://www.playframework.com/documentation/2.5.x/JavaTemplates#Escaping

TimMoore commented on web-gateway/app/views/searchItem.scala.html:
```diff
@@ -12,6 +12,8 @@
 
     <h2>@message("searchItems")</h2>
 
+	<p>@message("searchInstructions")</p>
```
Please be sure to use spaces rather than tabs for indentation.

If you're using Eclipse, here is a guide for configuring it to use spaces: http://stackoverflow.com/a/408449/29470

mehulsr replied:
Thanks, will fix this in the next commit. 
I'm not using any IDE at the moment, but the Atom Editor. Is it advisable to use an IDE?

mehulsr replied:
Fixed this is new commit.

ignasi35 commented on web-gateway/conf/messages:
```diff
@@ -85,3 +85,6 @@ invalid.duration=Please enter a valid duration
 createUserInstructions=Here you can create a new user who can participate in and create auctions.
+searchInstructions=Here you can search for auctioned items. Search functionality can be enabled by downloading and running a local instance of elasticsearch. \
+	Steps for doing this can be found in README.md.\
+	You can filter your search by specifying the maximum price of items to search for and the currency to search in.
```
Hi @mehulsr, thanks for this contribution. 
I have a question: are the newlines preserved when rendering this message in HTML? Or put another way, will this message be rendered as 3 `<p>` blocks? If not, is there 

I think we could rewrite them so that the generic instruction on how to use the feature appear first (since they always apply), and the details on how to get the feature up and running (install ES and/or check README) appear last. Also, I think it's worth mentioning that once the system is started, there's no need to restart it when installing and running ES since the search service will be able to recover.

mehulsr replied:
Hi @ignasi35 
The newlines are not preserved at the time of rendering in HTML. 
I shall reorder the instructions in the next commit to separate general search instructions and installation instructions.

mehulsr replied:
Fixed this is new commit.

TimMoore commented:
Do you still intend to add a link to `README.md` @mehulsr?

mehulsr commented:
@TimMoore Yes I think it would be appropriate. I'll try to do it one more time, else we can close the PR?

TimMoore commented:
OK no worries, just let us know when you're ready for us to merge it.

mehulsr commented:
Yes, sure :smiley:

mehulsr commented:
@TimMoore @ignasi35 
It took me a while to figure out, but I added link to the README.md file in search instructions. Currently, this link downloads the README.md file or shows it in the browser (if a Markdown plug in is enabled).
I have also updated the instructions naming convention as suggested in [this PR](https://github.com/lagom/online-auction-java/pull/66). 
Please review and suggest modifications.

ignasi35 commented on web-gateway/conf/routes:
```diff
@@ -22,3 +22,5 @@ GET         /search                     controllers.SearchController.searchForm(
 
 GET         /assets/*file               controllers.Assets.at(path = "/public", file)
+
+GET         /*file                      controllers.ExternalAssets.at(path="../", file)
```
hi @mehulsr we can't do this. With this route I can download any file in the folder including the source code (test http://localhost:9000/LICENSE or http://localhost:9000/build.sbt for example).

I think we can safely link to https://github.com/lagom/online-auction-java/blob/master/README.md  on the instructions (in which case this route won't be necessary).

mehulsr replied:
Ah, yes that's true. Okay, pushing with this change now

mehulsr commented:
@TimMoore @ignasi35 Please review the changes

ignasi35 commented:
Looks great! Thanks @mehulsr
